examples/periodic_tasks/src/main.rs

```diff
@@ -172,26 +171,53 @@ fn main() -> ! {
         // --- Test critical end ---
 
         unsafe {
+            let mut cycle_hi: u32;
+            let mut cycle_lo: u32;
+            let mut minstret_hi: u32;
+            let mut minstret_lo: u32;
+
+            core::arch::asm!("csrr {0}, 0xB00", out(reg) cycle_lo);
+            core::arch::asm!("csrr {0}, 0xB80", out(reg) cycle_hi);
+            core::arch::asm!("csrr {0}, 0xB02", out(reg) minstret_lo);
+            core::arch::asm!("csrr {0}, 0xB82", out(reg) minstret_hi);
+
+            sprintln!("cycles: {}{}", cycle_hi, cycle_lo);
+            sprintln!("instrs: {}{}", minstret_hi, minstret_lo);
+
+            /* Both of these alternatives cause an illegal instruction
+
+            //let cycle_lo = riscv::register::cycle::read();
+            //let cycle_hi = riscv::register::cycleh::read();
+            //let minstret_lo = riscv::register::minstret::read();
+            //let minstret_hi = riscv::register::minstret::read();
+            //-----------------------------------------------------
+            let cycle = riscv::register::cycle::read64();
+            let minstret = riscv::register::minstret::read64();
+
+            sprintln!("cycles: {}", cycle );
+            sprintln!("instrs: {}", minstret);
+            */
             sprintln!(
                 "Task counts:\r\n{} | {} | {} | {}",
                 TASK0_COUNT,
                 TASK1_COUNT,
                 TASK2_COUNT,
                 TASK3_COUNT
             );
-            let total_in_task0 = TASK0.duration_us * TASK0_COUNT as u32;
-            let total_in_task1 = TASK1.duration_us * TASK1_COUNT as u32;
-            let total_in_task2 = TASK2.duration_us * TASK2_COUNT as u32;
-            let total_in_task3 = TASK3.duration_us * TASK3_COUNT as u32;
+            let total_in_task0 = TASK0.duration_ns * TASK0_COUNT as u32;
+            let total_in_task1 = TASK1.duration_ns * TASK1_COUNT as u32;
+            let total_in_task2 = TASK2.duration_ns * TASK2_COUNT as u32;
+            let total_in_task3 = TASK3.duration_ns * TASK3_COUNT as u32;
             sprintln!(
-                "Theoretical total duration spent in task workload (us):\r\n{} | {} | {} | {} = {}",
+                "Theoretical total duration spent in task workload (ns):\r\n{} | {} | {} | {} = {}",
                 total_in_task0,
                 total_in_task1,
                 total_in_task2,
                 total_in_task3,
                 total_in_task0 + total_in_task1 + total_in_task2 + total_in_task3,
             );
 
+            /* TODO: reintroduce after mtimer div fixed
```

Comment:
Yep. I think for now it should be OK to just change the `const PERIPH_CLK_DIV: u32 = 2;` to 1 from `atalanta_bsp/src/mtimer.rs` and `atalanta_bsp/src/timer_group.rs`. Then re-introduce these.